### DISCUSSION TITLE
Experimenting on 09_strings/strings1.rs

### DISCUSSION THREAD

**Discussion by neerajnangiredddy** [rust-lang/rustlings#2184] [Q&A] [answered]
```rust
fn current_favorite_color() -> str {
    "blue"
}

fn main() {
    let answer = current_favorite_color();
    println!("My current favorite color is {answer}");
}
```

why the above code doesn't work?. 
why can't the compiler know the size of hard coded string "blue".

**Comment by mo8it** [comment, marked as answer]
String literals like `"blue"` have the type `&str`. You can think of them as being borrowed from a fixed space in memory, but they are valid until the program terminates. You need to change the return type to `&str`.
